### PR TITLE
[NEAT-595, NEAT-600] Improve neat to cdf data model

### PR DESCRIPTION
**Pull request by doctrino**

### CODE REVIEW

File: cognite/neat/_rules/exporters/_rules2dms.py [hunk 1]
```diff
@@ -306,50 +307,38 @@ def _categorize_items_for_upload(
...
-            elif is_redeploying:
-                to_update.append(item)
-                to_delete.append(cdf_item)
+                categorized.to_create.append(item)
+            elif is_redeploying or self.existing == "recreate":
+                if loader.has_data(cdf_item) and not self.drop_data:
+                    categorized.to_skip.append(cdf_item)
+                else:
+                    categorized.to_delete.append(cdf_item.as_write())
+                    categorized.to_create.append(item)
```
Comment (Juliamg) [inline cognite/neat/_rules/exporters/_rules2dms.py:316]: should item also be `.as_write` ?

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ Changes are grouped as follows:
 ### Changed
 - NeatEngine version to the latest release that has a breaking change to the interface; `2.0.0`
 - NeatEngine version `2.0.1` with bugfix for getting catalog file from zipped folder
+- [BREAKING ]The `neat.to.cst.data_model(...)` has been reworked. The new method has new parameters and new
+  behavior.
```
Comment (Juliamg) [inline docs/CHANGELOG.md:33]: Could you add a sentence on what the new behavior is?

**Review by Juliamg — APPROVED** [review]
I just have two comments. Otherwise it looks good and readable

File: cognite/neat/_rules/exporters/_rules2dms.py (hunk 1 above)
Comment (doctrino) [reply cognite/neat/_rules/exporters/_rules2dms.py:316]: no, the item is the local one which is already an write item.

**Pull request merged**